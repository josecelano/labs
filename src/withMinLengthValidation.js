import React from 'react';

const withMinLengthValidation = (Component, minLength) => {
  const warning = <span>Minimum {minLength} required.</span>;
  return (props) => {
    return <div><Component {...props} />{ props.value.length < minLength ? warning : ''}</div>
  }
}

export default withMinLengthValidation