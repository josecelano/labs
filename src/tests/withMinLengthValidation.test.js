import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withMinLengthValidation from '../withMinLengthValidation';

const Input = (props) => (
  <input {...props} />
)

const InputWithValidation = withMinLengthValidation(Input, 4)

it('should show a warning if a string value has less then 4 characters', () => {
  const tree = toJson(shallow(<InputWithValidation value={'hi'} />));
  expect(tree).toMatchSnapshot();
});

it('should show a warning if a number value has less then 4 digits', () => {
  const tree = toJson(shallow(<InputWithValidation value={42} />));
  expect(tree).toMatchSnapshot();
});

it('should show no warning if a string value has more then 4 characters', () => {
  const tree = toJson(shallow(<InputWithValidation value={'Ada Lovelace'} />));
  expect(tree).toMatchSnapshot();
});
