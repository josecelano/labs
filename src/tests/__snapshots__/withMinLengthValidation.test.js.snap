exports[`test should show a warning if a number value has less then 4 digits 1`] = `
<div>
  <Component
    value={42} />
  <span>
    Minimum 
    4
     required.
  </span>
</div>
`;

exports[`test should show a warning if a string value has less then 4 characters 1`] = `
<div>
  <Component
    value="hi" />
  <span>
    Minimum 
    4
     required.
  </span>
</div>
`;

exports[`test should show no warning if a string value has more then 4 characters 1`] = `
<div>
  <Component
    value="Ada Lovelace" />
</div>
`;
